### PR TITLE
add support for batchScheduler in Spark Tasks

### PR DESCRIPTION
**Pull request by migueltol22**
Signed-off-by: Miguel <mtoledo@lyft.com>

# TL;DR
Spark has support for batchScheduler such as Volcano however even if it enabled in the spark operator it cannot currently be used because we don't set it as part of the sparkApp.

Here we allow a user to specify `spark.batchScheduler` as part of their config to leverage the batchScheduler option. This can also be enabled for all spark tasks as part of the default spark config an admin sets in the plugins config

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: go/tasks/plugins/k8s/spark/spark_test.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ var (
 		"spark.flyte.feature1.enabled":    "true",
 		"spark.flyteorg.feature2.enabled": "true",
 		"spark.flyteorg.feature3.enabled": "true",
+		"spark.batchScheduler":            "volcano",
```
Comment (kumare3) [inline go/tasks/plugins/k8s/spark/spark_test.go:50]: why add it to spark-conf? Can we add it to just regular config? And I would prefer the config to say
scheduler: `default` - as default
and users can then suplement it as `scheduler`: `volcano`

Comment (kumare3) [reply go/tasks/plugins/k8s/spark/spark_test.go:50]: cc @akhurana001

Comment (migueltol22) [reply go/tasks/plugins/k8s/spark/spark_test.go:50]: Reason for spark conf is that for our use case we don't want to enable it for everyone, we'd like to try it out with a specific workflow. Adding it to regular config will set it for all unless we move to a config for each project/workflow. And I have no preference on whether `batchScheduler` or `scheduler`. Spark uses batch scheduler so figured I'd keep the same naming convention
